### PR TITLE
fix: add word boundray to check markdown italic

### LINKED ISSUE #40787: Markdown incorrect styling in editor
- VSCode Version: 1.19.1

In Markdown files, the editor is incorrectly styling every other inner word surrounded by underscores as italicized. The preview correctly shows them as not italicized.

markdown-italic-bug

### PR DESCRIPTION
**Pull request by mikaoelitiana**
fixes #40787

### CODE REVIEW

File: extensions/markdown/syntaxes/markdown.tmLanguage [hunk 1]
```diff
@@ -3318,7 +3318,7 @@
 				<dict>
 					<key>begin</key>
 					<string>(?x)
-								(\*|_)(?=\S)								# Open
```
Comment (mjbvz) [inline extensions/markdown/syntaxes/markdown.tmLanguage]: Can you please update `markdown.tmLanguage.base` as well. We use that file to generate this one. You can run the generator using `(cd extensions/markdown; npm run update-grammar)`

Comment (electriccode) [reply extensions/markdown/syntaxes/markdown.tmLanguage]: line 3371 must be changed from 
`<string>(?&lt;=\S)(\1)((?!\1)|(?=\1\1))</string>`
to
`<string>(?&lt;=\S)(\1)((?!\1)|(?=\1\1))\b</string>`

Comment (mikaoelitiana) [reply extensions/markdown/syntaxes/markdown.tmLanguage]: Thank you for review. I will fix that today

**Comment by mjbvz** [comment]
Thanks! This will be in the next insiders build and is on track for VS Code 1.20

**Comment by mikaoelitiana** [comment]
Great news!!! Thank you

Mika ANDRIANARIJAONA
mikaoelitiana@gmail.com
06 95 08 91 24
@mikaoelitiana

2018-01-24 17:41 GMT+01:00 Matt Bierner <notifications@github.com>:

**Pull request merged**